control-plane/api/mesh/v2beta1/gateway_class_config_types.go

```diff
@@ -31,66 +31,98 @@ type GatewayClassConfig struct {
+	Role GatewayClassRoleConfig `json:"role,omitempty"`
+	// RoleBinding contains config specific to the RoleBinding created from this GatewayClass
+	RoleBinding GatewayClassRoleBindingConfig `json:"roleBinding,omitempty"`
+	// Service contains config specific to the Service created from this GatewayClass
+	Service GatewayClassServiceConfig `json:"service,omitempty"`
+	// ServiceAccount contains config specific to the corev1.ServiceAccount created from this GatewayClass
+	ServiceAccount GatewayClassServiceAccountConfig `json:"serviceAccount,omitempty"`
+}
 
-	// +kubebuilder:validation:Enum=ClusterIP;NodePort;LoadBalancer
-	ServiceType *corev1.ServiceType `json:"serviceType,omitempty"`
+type GatewayClassDeploymentConfig struct {
+	GatewayClassAnnotationsAndLabels `json:",inline"`
+
+	// Container contains config specific to the created Deployment's container
+	Container GatewayClassContainerConfig `json:"container,omitempty"`
+	// InitContainer contains config specific to the created Deployment's init container
+	InitContainer GatewayClassInitContainerConfig `json:"initContainer,omitempty"`
+	// NodeSelector specifies the node selector to use on the created Deployment
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// PriorityClassName specifies the priority class name to use on the created Deployment
+	PriorityClassName string `json:"priorityClassName"`
+	// SecurityContext specifies the security context for the created Deployment's Pod
+	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
+	// Tolerations specifies the tolerations to use on the created Deployment
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
 
-	// NodeSelector is a selector which must be true for the pod to fit on a node.
-	// Selector which must match a node's labels for the pod to be scheduled on that node.
-	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
-	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+type GatewayClassInitContainerConfig struct {
+	// Consul specifies configuration for the consul-k8s-control-plane init container
+	Consul GatewayClassConsulConfig `json:"consul,omitempty"`
+	// Resources specifies the resource requirements for the created Deployment's init container
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+}
 
-	// Tolerations allow the scheduler to schedule nodes with matching taints.
-	// More Info: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
-	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+type GatewayClassContainerConfig struct {
+	// Consul specifies configuration for the consul-dataplane container
+	Consul GatewayClassConsulConfig `json:"consul,omitempty"`
+	// Resources specifies the resource requirements for the created Deployment's container
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Is this where things like meshGateway.replicas would get set?

Reply:
This draft doesn't have all fields set yet. I'd expect it to go where you linked to 💯

Reply:
Add replicas configuration that looks like
```yaml
deployment:
  replicas:
    default: 3
    min: 2
    max: 8
```